### ISSUE TITLE
"Failed" clicking/collecting on facebook

### ISSUE THREAD

**Issue by BunnyTheLifeguard** [dhowe/AdNauseam#1328]
### Describe the issue

Somehow AN stopped working for me on facebook. It now seems to be stuck on 355 ads detected, but does not click them or detect new ones.
AN works fine now on other sites like the top 15 most popular news websites, but seems to be "frozen" on facebook for me.

### Screenshot in which the issue can be seen

![screen shot 2018-02-13 at 16 48 27](https://user-images.githubusercontent.com/36411819/36159531-3bf56d00-10df-11e8-88d0-6675aefaa5b2.png)

### Your settings

- OS/version: OS X 10.13.3 (High Sierra)
- Browser/version: Waterfox 56.0.4.1
- AdNauseam version: 3.5.101
- Other extensions you have installed: BetterTTV, Blur, Cookie AutoDelete, Decentraleyes, DeepLT, Disable HTML5 Autoplay, HTTPS Everywhere, KeePaddHttp-Connector, NoFap Panic Button, NoScript (set to allow globally), Private Tab, RandomUA, Reddit Enhancement Suite, Tampermonkey, TrackMeNot

**Comment by BunnyTheLifeguard** [comment]
Quick update: I got rid of the addon **RandomUA** and instead installed **Canvas Fingerprint Defender**. Now it seems to at least detect new ads again. It also clicks every ad but the "l.facebook.com" ones.

![screen shot 2018-02-14 at 23 52 13](https://user-images.githubusercontent.com/36411819/36232570-9d568120-11e2-11e8-9e7e-0d8bf252f316.png)

**Comment by cqx931** [comment]
Youtube:
As most of the ads from youtube are video ads, adnauseam does help you to block them, but there is no image ad to collect. The number we are showing in the menu is actually the number of ads we collected.

@dhowe  Maybe we should change that text to "XX Ads collected" to be less confusing?

**Comment by dhowe** [comment]
Yes, that makes sense to me, @cqx931. But pls check that the added width doesn't mess up the display on any platforms...

**Comment by BunnyTheLifeguard** [comment]
Okay, due to switching back to firefox (from waterfox), I tested a couple of settings in about:config from privacytools.io and here is what I found:

`privacy.resistFingerprinting = true` (This one seems to cause the layout issues)

`browser.cache.offline.enable = false` (This one shows the wrong (all) numbers of ads detected)

Hope that helps.

**Comment by cqx931** [comment]
Thank you for the feedbacks! @BunnyTheLifeguard 
I just verified that `privacy.resistFingerprinting = true` does influence the layout in menu.
However I can't verify the second setting. I tested it with both image ads and text ads with only adnauseam installed and the number seems quite correct to me. Could you please show us an example of your wrong number?(which site, only adnauseam installed, empty the gallery beforehand, the number you see from adnauseam menu, the number you are expecting to see)

Firefox version: 59.0b7

**Comment by cqx931** [comment]
The layout issue is fixed here: https://github.com/dhowe/AdNauseam/pull/1332

**Comment by BunnyTheLifeguard** [comment]
@cqx931 You're right, the second one was not causing this issue.
Can't really pin it down to a specific thing. Sometimes it works showing the ads normally, sometimes it doesn't. We're still talking about facebook. 🤔

**Comment by BunnyTheLifeguard** [comment]
@cqx931 You were right, the second setting was not causing the bug.
After playing around with it again it's working fine with the only exception being facebook, but there it just happens when the newsfeed is set to "Most Recent", newsfeed set to "Top Stories" is working fine for now.

Here is a screenshot for the "Most Recent" setting:

![screen shot 2018-02-23 at 20 42 38](https://user-images.githubusercontent.com/36411819/36618604-0c1b2344-18ec-11e8-8bb0-ae8ea12da77f.png)

**Comment by BunnyTheLifeguard** [comment]
Okay, since today it also works again with "Most Recent"… I'm confused. 😵

**Comment by mneunomne** [comment]
Outdated. all facebook are dynamic created and changing all the time so its not possible to keep up with the changes. Closing for now.
